# **Şəkil 1**

# **1. Task**

### **English question**

- ### Write a program that reads multi-digit numbers until 0 and calculates the sum of the digit 2 of the input numbers.

### **Azerbaycanca**

- ### 0 daxil edilənə qədər çoxrəqəmli ədədləri oxuyun və daxil edilən bütün ədədlərin **2 rəqəminin cəmini** hesablayın.

In [66]:
total = 0
while True:
    s = input()
    if s == "0":
        break
    if '2' in s:
        total += 2 * s.count('2')   
print(total)


512345
725525
125327
0
10


# **2. Task**

### **English question**
- ### Write a program that reads numbers separated by spaces until EOF and counts the number of the input numbers that end in 5, line by line.

### **Azerbaycanca**
- ### Sətir-sətir verilmiş, aralarında boşluq olan ədədləri oxuyun və hər sətirdə **5 rəqəmi ilə bitən** ədədlərin sayını çap edin.


In [8]:
while True:
    try:
        line = input()
    except EOFError:
        break

    nums = line.split()
    count = 0
    for n in nums:
        if n.endswith("5"):
            count += 1
    print(count)



12
0

0


KeyboardInterrupt: Interrupted by user

In [65]:
while True:
    line = input()
    if line == '0':
        break

    nums = line.split()
    count = 0
    for n in nums:
        if n.endswith("5"):
            count += 1
    print(count)

25 16 5 36
2
23
0
45 7 85 15
3
0


# **3. Task**

### **English question**

- ### Write a program that reads strings until the string "END" and replace each consonant with an "*" character.

### **Azerbaycanca**

- ### "END" sözü daxil edilənə qədər bütün sözləri oxuyun və bütün **samit hərfləri** `*` işarəsi ilə əvəz edin.




In [28]:
vowels = "aeiouAEIOU"

while True:
    s = input()
    if s == "END":
        break

    res = ""
    for ch in s:
        if ch.isalpha() and ch not in vowels:
            res += "*"
        else:
            res += ch
    print(res)


apple
a***e
pear
*ea*
end
e**
END


# **4. Task**

### **English question**

- ### Write a program that randomly generates **n integers** between 10 and 70 and determines the **geometric mean** of the numbers divisible by 3.
- ### Print the result to three decimal places.
- ### If there are no such numbers, print 0.

### **Azerbaycanca**

- ### 10 ilə 70 arasında təsadüfi olaraq **n ədəd tam ədəd** yaradın. İçərisində **3-ə bölünən** ədədlərin **həndəsi ortasını** hesablayın.
- ### Nəticəni 3 onluq dəqiqliklə çap edin.
- ### Əgər belə ədəd yoxdursa, 0 çıxarın.

In [29]:
    import random
    import math

    n = int(input("n = "))
    nums = [random.randint(10, 70) for _ in range(n)]
    print("Generated:", *nums)

    filtered = [x for x in nums if x % 3 == 0]

    if len(filtered) == 0:
        print(0)
    else:
        product = 1
        for x in filtered:
            product *= x
        geo_mean = product ** (1/len(filtered))
        print(f"{geo_mean:.3f}")


n = 5
Generated: 29 41 61 36 47
36.000


# **Şəkil 2**

# **1. Task

- ### **English:**
- ### Write a program that reads multi-digit numbers until 0 and returns the product of the even digits of the input number.

- ### **Azerbaijani:**
- ### 0-a qədər çoxrəqəmli ədədləri oxuyan və daxil edilən ədədlərin cüt rəqəmlərinin hasilini qaytaran proqram yazın.

In [30]:
result = 1
while True:
    n = int(input())
    if n == 0:
        break

    product = 1
    has_even = False
    for digit in str(n):
        if int(digit) % 2 == 0:
            product *= int(digit)
            has_even = True

    if has_even:
        result *= product
    else:
        result *= 0  
print(result)


12345
725
1237
0
32


# 2. Task

- ### **English:**
- ### Write a program that reads numbers separated by spaces until the EOF and counts the elements which are square numbers (1, 4, 9, 16, 25, …).

- ### **Azerbaijani:**
- ### Mətn sonuna qədər (EOF) boşluqla ayrılmış ədədləri oxuyan və kvadrat ədədlərin sayını hesablayan proqram yazın.

In [36]:
import math
while True:
    line = input()
    if line == "0":  
        break
    
    count = 0
    nums = map(int, line.split())
    for n in nums:
        if int(math.sqrt(n)) ** 2 == n:
            count += 1

    print(count)


2 16 5 36
2
25
1
4 7 81 1
3
0


# **3. Task

- ### **English:**
- ### Write a program that reads strings until the "END" string and deletes the space characters.

- ### **Azerbaijani:**
- ### "END" sözünə qədər sətirləri oxuyan və sətirlərdən boşluqları silən proqram yazın.

In [37]:
while True:
    s = input()
    if s == "END":
        break
    print(s.replace(" ", ""))
print()


apple tr      ee
appletree
prog ram     ming
programming
END



# **4. Task

- ### **English:**
- ### Write a program that randomly generates n elements between 15 and 65, counts the arithmetic mean of the even values, and displays the result with two decimal places.

- ### **Azerbaijani:**
- ### 15 və 65 arasında n element təsadüfi yaradan, cüt ədədlərin arifmetik ortasını hesablayan və nəticəni iki onluq yerlə göstərən proqram yazın.

In [39]:
import random

n = int(input())
numbers = [random.randint(15, 65) for _ in range(n)]
evens = [x for x in numbers if x % 2 == 0]
print(*evens)

if evens:
    mean = sum(evens) / len(evens)
    print(f"{mean:.2f}")
else:
    print("0.00")


5
28 36 60
41.33


# **Şəkil 3**

# **1. Task**

- ### **English:**
- ### Write a program that reads multi-digit numbers (≥10) until 0 and returns the sum of the digits of the smallest input number.

- ### **Azerbaijani:**
- ### 0-a qədər çoxrəqəmli ədədləri (≥10) oxuyan və daxil edilmiş ədədlərdən ən kiçiyinin rəqəmlərinin cəmini qaytaran proqram yazın.

In [49]:
numbers = []
while True:
    n = int(input())
    if n == 0:
        break
    if n >= 10:
        numbers.append(n)

if numbers:
    smallest = min(numbers)
    digit_sum = sum(int(d) for d in str(smallest))
    print(digit_sum)
else:
    print(0)

12345
785
4236
0
20


# **2. Task**

- ### **English:**
- ### Write a program that reads three ascending numbers separated by spaces until EOF and prints YES if they form a -### Pythagorean triplet (`a² + b² = c²`) and NO otherwise.

- ### **Azerbaijani:**
- ### EOF-a qədər artan sıra ilə 3 ədədi oxuyan və əgər ədədlər Pifaqor üçlüyünü (`a² + b² = c²`) təşkil edirsə YES, əks halda NO çap edən proqram yazın.

In [67]:
while True:
    line = input()
    if line == "0":  
        break

    a, b, c = line.split()
    if int(a)**2 + int(b)**2 == int(c)**2:
        print("YES")
    else:
        print("NO")

3 4 5
YES
2 6 8
NO
7 24 25
YES
0


# **3. Task**

- ### **English:**
- ### Write a program that reads strings until the "END" string and deletes all numeric characters.

- ### **Azerbaijani:**
- ### "END" sözünə qədər sətirləri oxuyan və sətirlərdən bütün rəqəmləri silən proqram yazın.

In [70]:
while True:
    s = input()
    if s == "END":
        break
    result = ''.join(ch for ch in s if not ch.isdigit())
    print(result)


Af1Pa2Pb4LrE
AfPaPbLrE
END


# **4. Task**

- ### **English:**
- ### Write a program that randomly generates n numbers between 50 and 100, calculates the mean of the numbers divisible by 4, and displays the result with two decimal places. If there are no such numbers, return 0.

- ### **Azerbaijani:**
- ### 50 və 100 arasında n təsadüfi ədəd yaradan, 4-ə bölünən ədədlərin ortasını hesablayan və nəticəni iki onluq yerlə göstərən proqram yazın. Əgər belə ədəd yoxdursa, 0 qaytarın.

In [74]:
import random

n = int(input("n = "))
numbers = [random.randint(50, 100) for _ in range(n)]
div4 = [x for x in numbers if x % 4 == 0]
print(*div4)

if div4:
    mean = sum(div4) / len(div4)
    print(f"{mean:.2f}")
else:
    print("0")

n = 5
76 68 64
69.33


# **Şəkil 4**

# **1. Task**

- ### **English:**
- ### Write a program that reads multi-digit numbers until 0 and calculates the difference between the sum of even and odd digits of the input numbers.

- ### **Azerbaijani:**
- ### 0 daxil edilənədək çoxrəqəmli ədədləri oxuyun və həmin ədədlərin cüt rəqəmlərinin cəmi ilə tək rəqəmlərinin cəminin fərqini hesablayın.



In [76]:
while True:
    n = int(input())
    if n == 0:
        break

    even_sum = 0
    odd_sum = 0

    for d in str(n):
        d = int(d)
        if d % 2 == 0:
            even_sum += d
        else:
            odd_sum += d

    print(even_sum - odd_sum)

12345
-3
24680
20
13579
-25
0


# **2. Task**

- ### **English:**

- ### Write a program that reads integers separated by spaces until EOF and calculates the sum of the elements divisible by 9 in each row.

- ### **Azerbaijani:**

- ### EOF-a qədər boşluqla ayrılmış tam ədədləri oxuyun və hər sətirdə 9-a bölünən ədədlərin cəmini hesablayın.

In [81]:
while True:
    line = input()
    if line == '0':
        break
    nums = map(int, line.split())
    s = sum(x for x in nums if x % 9 == 0)
    print(s)        


45  2  25  9  4
54
7
0
9 18
27
0


# 3. Task

- ### **English:**

- ### Write a program that reads strings until the word "STOP" and replaces all vowel characters with '*'.

- ### **Azerbaijani:**

- ### STOP" sözünə qədər sətirləri oxuyun və sətirlərdəki bütün saitləri `*` simvolu ilə əvəzləyin.

In [82]:
vowels = "aeiouAEIOU"

while True:
    s = input()
    if s == "STOP":
        break
    result = "".join("*" if ch in vowels else ch for ch in s)
    print(result)    
    

AbCdEfGh
*bCd*fGh
HeHeHe
H*H*H*
STOP


In [84]:
vowels = "aeiouAEIOU"   
while True:
    s = input()
    if s == "STOP":
        break

    res = ""
    for ch in s:
        if ch.isalpha() and ch in vowels:
            res += "*"
        else:
            res += ch
    print(res) 

AbCdEfGh
*bCd*fGh
HeHeHe
H*H*H*
STOP


# **4.Task

- ### **English:**

- ### Write a program that randomly generates **n integers between 10 and 100**, prints these numbers, finds the three numbers divisible by 3, calculates their geometric mean (( (a \cdot b \cdot c)^{1/3} )), and prints it to three decimal places. If fewer than 3 such numbers exist, print 0.

### **Azerbaijani:**

- ### 10 və 100 arasında **n ədəd** təsadüfi yaradan, onları ekrana çıxaran, 3-ə bölünən ilk üç ədədi seçən, onların həndəsi ortasını (( (a \cdot b \cdot c)^{1/3} )) hesablayan və üç onluqla çap edən proqram yazın. Əgər belə ədəd 3-dən azdırsa, 0 yazın.

In [91]:
import random
n = int(input("n = "))
nums = [random.randint(10, 100) for _ in range(n)]
print(*nums)

div3 = [x for x in nums if x % 3 == 0]

if len(div3) >= 3:
    a, b, c = div3[:3]
    geo = (a * b * c) ** (1/3)
    print(f"{geo:.3f}")
else:
    print(0)

n = 15
20 21 54 82 68 25 40 77 95 12 25 71 93 35 92
23.874



# PDF


# ✅ **1) Ex**

- ### **English:**

- ### Write a program that reads multi-digit numbers until **0** and returns the **product of the even digits** of the input numbers.

- ### **Azerbaijani:**

- ### 0 daxil edilənədək çoxrəqəmli ədədləri oxuyan və daxil edilmiş ədədlərin **cüt rəqəmlərinin hasilini** qaytaran proqram yazın.

-  12345 
-  725 
-  1257 

- → 32

In [8]:
result = 1
while True:
    n = int(input())
    if n == 0:
        break

    product = 1
    has_even = False
    for digit in str(n):
        if int(digit) % 2 == 0:
            product *= int(digit)
            has_even = True

    if has_even:
        result *= product
    else:
        result *= 1  
print(result)

12345
725
1237
0
32


In [9]:
total = 1

while True:
    n = input()
    if n == "0":
        break

    p = 1
    found = False
    for d in n:
        if d in "02468":
            p *= int(d)
            found = True

    if found:
        total *= p
    else:
        total *= 1

print(total)


12345
725
1237
0
32


# ✅ **1) Task**

- ### **English:**

- ### Write a program that reads multi-digit numbers until the EOF and counts the number of digit **'4'** in each input number.

- ### **Azerbaijani:**

- ###  EOF-a qədər çoxrəqəmli ədədləri oxuyun və hər bir ədəddəki **'4' rəqəmlərinin sayını** hesablayın.


**Input:**

```
142341
482
524436
```

**Output:**

```
3
1
2
```

---



In [13]:

while True:
    n = input()
    if n == "0":
        break
    print(n.count("4"))


142341
2
482
1
524436
2
0


In [14]:
list_1 = []
while True:
    n = input()
    if n == "0":
        break
    list_1.append(n.count("4"))
for i in list_1:
    print(i)

142341
482
524436
0
2
1
2
